docs/building-modules/handler.md

```diff
@@ -69,20 +63,19 @@ ctx.EventManager().EmitEvent(
 
 These `events` are relayed back to the underlying consensus engine and can be used by service providers to implement services around the application. Click [here](../core/events.md) to learn more about `events`. 
 
-Finally, the `handler` function returns a `*sdk.Result` which contains the aforementioned `events` and an optional `Data` field. 
+Finally, the invoked `msgServer` method returns a `proto.Message` response and an `error` which are then wrapped in `*sdk.Result` which contains the aforementioned `events` and an optional `Data` field, using `sdk.WrapServiceResult` function. 
 
 +++ https://github.com/cosmos/cosmos-sdk/blob/d55c1a26657a0af937fa2273b38dcfa1bb3cff9f/proto/cosmos/base/abci/v1beta1/abci.proto#L81-L95
 
-Next is an example of how to return a `*Result` from the `gov` module:
-
-+++ https://github.com/cosmos/cosmos-sdk/blob/d55c1a26657a0af937fa2273b38dcfa1bb3cff9f/x/gov/handler.go#L67-L70
-
-For a deeper look at `handler`s, see this [example implementation of a `handler` function](https://github.com/cosmos/cosmos-sdk/blob/d55c1a26657a0af937fa2273b38dcfa1bb3cff9f/x/gov/handler.go) from the `gov` module.
-
 The `handler` can then be registered from [`AppModule.Route()`](./module-manager.md#appmodule) as shown in the example below:
 
 +++ https://github.com/cosmos/cosmos-sdk/blob/228728cce2af8d494c8b4e996d011492139b04ab/x/gov/module.go#L143-L146
 
+### Using [Legacy `Msg`s](messages-and-queries.md#legacy-msgs)
+
+In this case, `handler`s functions need to be implemented for each module `Msg` and should be used in `NewHandler` in the place of `msgServer` methods.
+`handler`s functions should return a `*Result` and an `error`.
```

Comment:
Maybe explain why someone would do that and what additional efforts he has if he is not using `MsgServer`.

Reply:
Should we actually drop this section? 
cc/ @clevinson @amaurymartiny @aaronc @robert-zaremba

Reply:
Added few more details in the new file `msg-services.md`, although I'm not sure why someone would actually do that.

Reply:
Because he could see some old examples with `handler*` functions